src/zcbor_common.c

```diff
@@ -246,3 +246,26 @@ size_t zcbor_header_len(size_t num_elems)
 		return 9;
 	}
 }
+
+
+int zcbor_entry_function(const uint8_t *payload, size_t payload_len,
+	void *result, size_t *payload_len_out, zcbor_state_t *state, zcbor_decoder_t func,
+	uint_fast32_t n_states, uint_fast32_t elem_count)
+{
+	zcbor_new_state(state, n_states, payload, payload_len, elem_count);
+
+	bool ret = func(state, result);
+
+	if (ret && (payload_len_out != NULL)) {{
+		*payload_len_out = MIN(payload_len,
+				(size_t)state[0].payload - (size_t)payload);
+	}}
+
+	if (!ret) {{
```

Comment:
This block can be moved just after  `ret` is instantiated and return early if `ret == false`. That will allow to remove `ret` from the `ret && (payload_len_out != NULL)` condition and just check if `payload_len_out` isn't `NULL`.